satellite/metainfo/metainfo.go

```diff
@@ -1395,14 +1646,103 @@ func (endpoint *Endpoint) DownloadSegment(ctx context.Context, req *pb.SegmentDo
 		return nil, status.Errorf(codes.Unauthenticated, err.Error())
 	}
 
-	// TODO implement logic
+	bucketID := createBucketID(keyInfo.ProjectID, streamID.Bucket)
+
+	exceeded, limit, err := endpoint.projectUsage.ExceedsBandwidthUsage(ctx, keyInfo.ProjectID, bucketID)
+	if err != nil {
+		endpoint.log.Error("retrieving project bandwidth total", zap.Error(err))
+	}
+	if exceeded {
+		endpoint.log.Sugar().Errorf("monthly project limits are %s of storage and bandwidth usage. This limit has been exceeded for bandwidth for projectID %s.",
```

Comment:
Why not use here a `log.Error` rather than the sugar version?

Reply:
the same answer as previously